### PR TITLE
setup automatic linting and add verbosity and quiet flags

### LINKED ISSUE #10: Add verbosity config
Kind has a `-v / --verbosity` flag that is used to control the info log verbosity.

We should expose it as a config - see https://github.com/engineerd/setup-kind/blob/46b89254fa0fc2e82cb94ce104518ac021f7f236/src/kind.ts#L31

### PR DESCRIPTION
**Pull request by mmorel-35**
Fixes #10 

This includes linting with eslint, prettier, lint-staged and husky

I also added the handling of verbosity and quiet flags from [here](https://github.com/kubernetes-sigs/kind/blob/main/pkg/cmd/kind/root.go)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ cluster (the Kind tools is configured in the path).
 
 Optional inputs:
 
+- `verbosity`: numeric log verbosity, (info = 0, debug = 3, trace = 2147483647) (default `"0"`)
+- `quiet`: silence all stderr output (default `"false"`)
```
Comment (radu-matei) [inline README.md:46]: Would you please mind adding these two flags at the end of the list?

File: action.yml [hunk 2]
```diff
@@ -5,6 +5,12 @@ inputs:
   version:
     description: "Version of Kind to use (default v0.11.1)"
     default: "v0.11.1"
+  verbosity:
+    description: "numeric log verbosity, (info = 0, debug = 3, trace = 2147483647)"
```
Comment (radu-matei) [inline action.yml:9]: nit: could you please keep the description a full upper-case sentence?

File: README.md (hunk 1 above)
Comment (mmorel-35) [reply README.md]: Ok, ! Done

File: src/kind.ts [hunk 3]
```diff
@@ -6,168 +6,215 @@ import * as artifact from '@actions/artifact';
...
-        this.image = image;
-        this.name = name;
-        this.waitDuration = waitDuration;
-        this.kubeConfigFile = kubeConfigFile;
-        this.skipClusterCreation = (skipClusterCreation == 'true');
-        this.skipClusterDeletion = (skipClusterDeletion == 'true');
-        this.skipClusterLogsExport = (skipClusterLogsExport == 'true');
-    }
+  verbosity: number;
+  quiet: boolean;
```
Comment (radu-matei) [inline src/kind.ts:24]: Could you please add the two inputs at the end, so the diff is clear?

File: src/main.ts [hunk 4]
```diff
@@ -15,11 +15,11 @@ async function run() {
 }
 
 function checkEnvironment() {
-  const supportedPlatforms : string[] = ["linux/x64"];
-  const platform : string = `${process.platform}/${process.arch}`;
+  const supportedPlatforms: string[] = ['linux/x64'];
+  const platform = `${process.platform}/${process.arch}`;
```
Comment (radu-matei) [inline src/main.ts:19]: Why remove the type annotation?

**Review by radu-matei — COMMENTED** [review]
LGTM, with a few really minor nits.
Thank you so much!

File: src/main.ts (hunk 4 above)
Comment (mmorel-35) [reply src/main.ts:19]: It's the application of [@typescript-eslint/no-inferrable-types](https://github.com/typescript-eslint/typescript-)eslint/blob/main/packages/eslint-plugin/docs/rules/no-inferrable-types.md

File: src/kind.ts (hunk 3 above)
Comment (mmorel-35) [reply src/kind.ts]: Done

File: action.yml (hunk 2 above)
Comment (mmorel-35) [reply action.yml]: What do you think about this new version : `Defines log verbosity with a numeric value, (info = 0, debug = 3, trace = 2147483647)` ?

File: src/main.ts (hunk 4 above)
Comment (radu-matei) [reply src/main.ts:19]: Ah, that makes sense.

**Review by radu-matei — APPROVED** [review]

**Pull request merged**